### PR TITLE
Add User Point in Datastore

### PR DESCRIPTION
**Pull request by salnad**
Add a points system in the Datastore to store a running total of points between rounds. Ran Java formatter on both files.

In **GameServlet**:
* add a new EmbeddedEntity field, `userPoints` to the Game Entity, acting as a Map between userId and point total
* create a function `createUserPoints` which instantiates a Map of User to points (starting at 0)
* TODO: update `createUserPoints` to add only users in the current "room" to the `userPoints` map, not all users

In **ChatServlet**
* Update `markUserGuessedCorrectly` to `markUserGuessedCorrectlyAndUpdatePoints` (so it handles updating points as well)
* Add a new private static constant `POINTS_PER_ROUND` to be used to determine how many points a user gets per round if they guess correctly (currently a hundred)

Video showing point values updating in Datastore (https://screencast.googleplex.com/cast/NTI1MjEwNTgxODM0MTM3NnxjNDg2MzM0ZS1kZA)

### CODE REVIEW

File: src/main/java/com/google/songgame/servlets/ChatServlet.java [hunk 1]
```diff
@@ -85,7 +86,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     if (checkIfUserPreviouslyGuessedCorrect(userId, currentRound)) {
```
Comment (guavafanclub) [inline src/main/java/com/google/songgame/servlets/ChatServlet.java:86]: rename userPreviouslyGuessedCorrectly, reads easier

File: src/main/java/com/google/songgame/servlets/ChatServlet.java [hunk 2]
```diff
@@ -85,7 +86,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
       messageType = "spectator";
     } else if (checkIfCorrectGuess(message, currentRound)) {
```
Comment (guavafanclub) [inline src/main/java/com/google/songgame/servlets/ChatServlet.java:88]: rename isCorrectGuess, reads easier

File: src/main/java/com/google/songgame/servlets/ChatServlet.java [hunk 3]
```diff
@@ -122,24 +123,33 @@ private String getUserId(HttpServletRequest request) throws IOException {
...
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+    EmbeddedEntity userGuessStatuses =
+        (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
     boolean userGuessStatus = (Boolean) userGuessStatuses.getProperty(userId);
     return userGuessStatus;
   }
 
-  private void markUserGuessedCorrectly(String userId, EmbeddedEntity currentRound, Entity currentGame) {
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+  private void markUserGuessedCorrectlyAndAddPoints(
```
Comment (guavafanclub) [inline src/main/java/com/google/songgame/servlets/ChatServlet.java:140]: add comments in this method

File: src/main/java/com/google/songgame/servlets/ChatServlet.java [hunk 4]
```diff
@@ -122,37 +123,49 @@ private String getUserId(HttpServletRequest request) throws IOException {
...
+    userPoints.setProperty(userId, updatedUserPoints);
+    
+    // Update both changes in Datastore
     currentGame.setProperty("currentRound", currentRound);
+    currentGame.setProperty("userPoints", userPoints);
     datastore.put(currentGame);
   }
 
-  private boolean checkIfCorrectGuess(String message, EmbeddedEntity currentRound) {
+  private boolean isCorrectGuess(String message, EmbeddedEntity currentRound) {
```
Comment (pruess-google) [inline src/main/java/com/google/songgame/servlets/ChatServlet.java:159]: Nice name change. I think the new name is a lot more readable at the sites where this method is called

File: src/main/java/com/google/songgame/servlets/ChatServlet.java [hunk 5]
```diff
@@ -122,37 +123,49 @@ private String getUserId(HttpServletRequest request) throws IOException {
...
-  private void markUserGuessedCorrectly(String userId, EmbeddedEntity currentRound, Entity currentGame) {
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+  private void markUserGuessedCorrectlyAndAddPoints(
+      String userId, EmbeddedEntity currentRound, Entity currentGame) {
+    // Changes the users "guess status" to show they've guessed correctly
+    EmbeddedEntity userGuessStatuses =
+        (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
     userGuessStatuses.setProperty(userId, true);
+    
+    // Increases the users points by a fixed amount
```
Comment (pruess-google) [inline src/main/java/com/google/songgame/servlets/ChatServlet.java:147]: We have plans to assign points based on how early a user guesses correctly, right (e.g. first correct guess gets most points, second correct guess gets second most points, etc)? If so, maybe add a TODO for this here

File: src/main/java/com/google/songgame/servlets/ChatServlet.java [hunk 6]
```diff
@@ -122,37 +123,49 @@ private String getUserId(HttpServletRequest request) throws IOException {
...
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+    EmbeddedEntity userGuessStatuses =
+        (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
     boolean userGuessStatus = (Boolean) userGuessStatuses.getProperty(userId);
     return userGuessStatus;
   }
 
-  private void markUserGuessedCorrectly(String userId, EmbeddedEntity currentRound, Entity currentGame) {
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+  private void markUserGuessedCorrectlyAndAddPoints(
```
Comment (pruess-google) [inline src/main/java/com/google/songgame/servlets/ChatServlet.java:140]: I think we have an opportunity to add tests around the guess checking methods: checkIfUserPreviouslyGuessedCorrect(), markUserGuessedCorrectlyAndAddPoints(), isCorrectGuess() if you refactor them into their own class (e.g. GuessCheck, GuessArbiter, etc). The only issue is that markUserGuessedCorrectlyAndAddPoints() writes to Datastore, so you could either:
1. (Easiest) Have the method that calls markUserGuessedCorrectlyAndAddPoints() within ChatServlet do the writing to datastore
2. (Harder) Mock out the dependency to datastore in your tests for the new class

I think these tests will become even more important as we make the points logic more complicated, which is why I'm suggesting this. Take a little time to look into this and let me know if you have any questions.

**Pull request merged**